### PR TITLE
利用 Keras 深度学习库进行词性标注教程

### LINKED ISSUE #3602: 利用 Keras 深度学习库进行词性标注教程
* 原文链接：[Part-of-Speech tagging tutorial with the Keras Deep Learning library](https://medium.com/@cdiscountdatascience/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library-d7f93fa05537)
  * Markdown文件：[文件地址](https://github.com/xitu/gold-miner/blob/master/TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md)
  * PR 地址：#3621
  * 文章分类：**深度学习**
  * **注意：文件位置在 `TODO1` 文件夹中，不是之前的 `TODO` 了。**
  ----
  * 翻译时间：2 天
  * 校对时间：1 天
  * 翻译积分：2 分
  * 校对积分：1 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by luochen1992**
译文翻译完成，resolve #3602

### CODE REVIEW

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 1]
```diff
@@ -32,42 +32,42 @@ CUSTOM_SEED = 42
 np.random.seed(CUSTOM_SEED)
 ```
 
-### Getting an annotated corpus
+### 获取已标注的语料库
 
-The [Penn Treebank](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.9.8216&rep=rep1&type=pdf) is an annotated corpus of POS tags. A sample is available in the `[NLTK](https://github.com/nltk/nltk)` python library which contains a lot of corpora that can be used to train and test some NLP models.
+[Penn Treebank](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.9.8216&rep=rep1&type=pdf) 是一个词性标注语料库。python库中有个示例 `[NLTK](https://github.com/nltk/nltk)` 包含能够用于训练和测试某些自然语言处理模型（NLP models）的语料库。
```
Comment (stormluke) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 「python库中有个示例」=>「Python 库中有个示例」

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 2]
```diff
@@ -87,17 +87,17 @@ This yields:
 'NNP', 'CD', 'RB', 'WDT', 'UH', 'NN', '-NONE-'}
 ```
 
-### Datasets preprocessing for supervised learning
+### 监督式学习的数据集预处理
 
-We split our tagged sentences into 3 datasets :
+我们将标记的句子划分成3个数据集 :
```
Comment (stormluke) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 「划分成3个数据集」=>「划分成 3 个数据集」

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 3]
```diff
@@ -87,17 +87,17 @@ This yields:
...
-*   a **validation dataset** used to tune the parameters of the classifier, for example to choose the number of units in the neural network,
-*   a **test dataset** used _only_ to assess the performance of the classifier.
+*   **训练集** 相当于拟合模型的样本数据，
+*   **验证集** 用于调整分类器的参数，例如选择网络中神经元的个数，
+*   **测试集** **仅**用于评估分类器的性能。
 
 ![](https://cdn-images-1.medium.com/max/800/1*DEDwnWZCWoaCAQQq5RG0xg.png)
 
-We use approximately 60% of the tagged sentences for training, 20% as the validation set and 20% to evaluate our model.
+我们使用大约 60% 的标记句子进行训练，20%作为验证集，20%用于评估我们的模型。
```
Comment (stormluke) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 注意下空格

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 4]
```diff
@@ -177,38 +177,38 @@ for pos_tags in tagged_sentences:
     return X, y
 ```
 
-For training, validation and testing sentences, we split the attributes into `X` (input variables) and `y` (output variables).
+对于训练、验证和测试句子，我们将属性分为“X”（输入变量）和“y”（输出变量）。
```
Comment (stormluke) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 「我们将属性分为“X”（输入变量）和“y”（输出变量）」=>「我们将属性分为 `X`（输入变量）和 `y`（输出变量）」

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 5]
```diff
@@ -177,38 +177,38 @@ for pos_tags in tagged_sentences:
...
 
-# Convert dict features to vectors
+# 将字典特征转换为向量
 X_train = dict_vectorizer.transform(X_train)
 X_test = dict_vectorizer.transform(X_test)
 X_val = dict_vectorizer.transform(X_val)
 ```
 
-Our `y` vectors must be encoded. The output variable contains 49 different string values that are encoded as integers.
+我们的'y'向量必须被编码。输出变量包含49个不同的字符串值，它们被编码为整数。
```
Comment (stormluke) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 「我们的'y'向量必须被编码」=> 「我们的 `y` 向量必须被编码」

**Comment by stormluke** [comment]
校对完成

**Comment by mingxing47** [comment]
校对认领

**Comment by fanyijihua** [comment]
@mingxing47 妥妥哒 :beers:

**Comment by leviding** [comment]
@mingxing47 赶紧来校对啦

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 6]
```diff
@@ -2,28 +2,28 @@
...
 
 Photo by [Joao Tzanno](https://unsplash.com/photos/G9_Euqxpu4k?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText) on [Unsplash](https://unsplash.com/search/photos/stamp?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText)
 
-Part-of-Speech tagging is a well-known task in [Natural Language Processing](https://en.wikipedia.org/wiki/Natural_language_processing). It refers to the process of classifying words into their parts of speech (also known as words classes or lexical categories). This is a supervised learning approach.
+在 [自然语言处理](https://en.wikipedia.org/wiki/Natural_language_processing) 中，词性标注是一件众所周知的任务。它指的是将单词按词性分类（也称为词类或词性类别）。这是一种有监督的学习方法。
 
-[Artificial neural networks](https://en.wikipedia.org/wiki/Artificial_neural_network) have been applied successfully to compute POS tagging with great performance. We will focus on the Multilayer Perceptron Network, which is a very popular network architecture, considered as the state of the art on Part-of-Speech tagging problems.
+[人工神经网络](https://en.wikipedia.org/wiki/Artificial_neural_network) 已成功应用于词性标注，并且性能卓越。我们将重点关注多层感知器网络，这是一种非常流行的网络结构，被视为解决词性标注问题的最新技术。（译者注： 对于词性标注问题，RNN 有更好的效果）
```
Comment (mingxing47) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: [性能卓越] => [表现卓越]
这里应该没有 体现出性能的意思

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 7]
```diff
@@ -87,17 +87,17 @@ This yields:
...
 
-We split our tagged sentences into 3 datasets :
+我们将标记的句子划分成 3 个数据集 : 
 
-*   a **training dataset** which corresponds to the sample data used to fit the model,
-*   a **validation dataset** used to tune the parameters of the classifier, for example to choose the number of units in the neural network,
-*   a **test dataset** used _only_ to assess the performance of the classifier.
+*   **训练集** 相当于拟合模型的样本数据，
+*   **验证集** 用于调整分类器的参数，例如选择网络中神经元的个数，
+*   **测试集** **仅**用于评估分类器的性能。
```
Comment (mingxing47) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: [性能] => [表现]

**Comment by mingxing47** [comment]
@leviding @luochen1992  不好意思有点拖稿，校对完成

**Comment by leviding** [comment]
@luochen1992 可以来修改啦

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md (hunk 7 above)
Comment (luochen1992) [reply TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 评估性能似乎更顺畅。

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 8]
```diff
@@ -2,104 +2,104 @@
 > * 原文作者：[Cdiscount Data Science](https://medium.com/@cdiscountdatascience?source=post_header_lockup)
 > * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
 > * 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md](https://github.com/xitu/gold-miner/blob/master/TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md)
-> * 译者：
+> * 译者：[luochen](https://github.com/luochen1992)
 > * 校对者：
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 添加校对者信息

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 9]
```diff
@@ -2,104 +2,104 @@
...
+由 [Unsplash](https://unsplash.com/search/photos/stamp?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText) 的 [Joao Tzanno](https://unsplash.com/photos/G9_Euqxpu4k?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText) 拍摄
 
-Part-of-Speech tagging is a well-known task in [Natural Language Processing](https://en.wikipedia.org/wiki/Natural_language_processing). It refers to the process of classifying words into their parts of speech (also known as words classes or lexical categories). This is a supervised learning approach.
+在 [自然语言处理](https://en.wikipedia.org/wiki/Natural_language_processing) 中，词性标注是一件众所周知的任务。它指的是将单词按词性分类（也称为词类或词性类别）。这是一种有监督的学习方法。
 
-[Artificial neural networks](https://en.wikipedia.org/wiki/Artificial_neural_network) have been applied successfully to compute POS tagging with great performance. We will focus on the Multilayer Perceptron Network, which is a very popular network architecture, considered as the state of the art on Part-of-Speech tagging problems.
+[人工神经网络](https://en.wikipedia.org/wiki/Artificial_neural_network) 已成功应用于词性标注，并且表现卓越。我们将重点关注多层感知器网络，这是一种非常流行的网络结构，被视为解决词性标注问题的最新技术。（译者注： 对于词性标注问题，RNN 有更好的效果）
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 对于词性标注问题前面多了个空格

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 10]
```diff
@@ -2,104 +2,104 @@
...
 
-**Let’s put it into practice!**
+**让我们把它付诸实践!**
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 叹号用全角

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 11]
```diff
@@ -2,104 +2,104 @@
...
 
-In this post you will get a quick tutorial on how to implement a simple Multilayer Perceptron in Keras and train it on an annotated corpus.
+在本文中，你将获得一个关于如何在 Keras 中实现简单的多层感知器的快速教程，并在已标注的语料库上进行训练。
 
-### Ensuring reproducibility
+### 确保可重复性
 
-In order to be sure that our experiences can be achieved again we need to fix the random seed for reproducibility:
+为了保证我们的实验能够复现，我们需要设置一个随机种子:
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 冒号用全角

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 12]
```diff
@@ -2,104 +2,104 @@
...
 'RBS', 'LS', 'PRP$', 'WRB', 'JJS', '``', 'EX', 'POS', 'WP', 'VBN', '-LRB-', '-RRB-', 
 'FW', 'MD', 'VBG', 'TO', '$', 'NNS', 'NNPS', "''", 'VBD', 'JJR', ':', 'PDT', 'SYM', 
 'NNP', 'CD', 'RB', 'WDT', 'UH', 'NN', '-NONE-'}
 ```
 
-### Datasets preprocessing for supervised learning
+### 监督式学习的数据集预处理
 
-We split our tagged sentences into 3 datasets :
+我们将标记的句子划分成 3 个数据集 :
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 冒号前多了空格

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 13]
```diff
@@ -177,79 +177,79 @@ for pos_tags in tagged_sentences:
     return X, y
 ```
 
-For training, validation and testing sentences, we split the attributes into `X` (input variables) and `y` (output variables).
+对于训练、验证和测试句子，我们将属性分为 X （输入变量）和 y （输出变量）。
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 符号和英文之间不用空格，本句有两个多余空格

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 14]
```diff
@@ -177,79 +177,79 @@ for pos_tags in tagged_sentences:
...
 y_train = np_utils.to_categorical(y_train)
 y_test = np_utils.to_categorical(y_test)
 y_val = np_utils.to_categorical(y_val)
 ```
 
-### Building a Keras model
+### 建立 Keras 模型
 
-`[Keras](https://github.com/fchollet/keras/)` is a high-level framework for designing and running neural networks on multiple backends like `[TensorFlow](https://github.com/tensorflow/tensorflow/)`, `[Theano](https://github.com/Theano/Theano)` or `[CNTK](https://github.com/Microsoft/CNTK)`.
+`[Keras](https://github.com/fchollet/keras/)` 是一个高级框架，用于设计和运行神经网络，它拥有多个后端像是 `[TensorFlow](https://github.com/tensorflow/tensorflow/)`, `[Theano](https://github.com/Theano/Theano)` or `[CNTK](https://github.com/Microsoft/CNTK)`.
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: or 翻译一下吧，否则突兀

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 15]
```diff
@@ -343,52 +343,52 @@ plot_model_performance(
...
+模型性能随迭代次数的变化。
 
-After 2 epochs, we see that our model begins to overfit.
+两次迭代之后，我们发现模型过拟合。
 
-### Evaluating our multilayer perceptron
+### 评估多层感知器
 
-Since our model is trained, we can evaluate it (compute its accuracy):
+由于我们模型已经训练好了，所以我们可以直接评估它:
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 冒号用全角

File: TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md [hunk 16]
```diff
@@ -343,52 +343,52 @@ plot_model_performance(
...
-*   `Keras`: The Python Deep Learning library_: [[doc]](https://keras.io/)
-*   Adam: A Method for Stochastic Optimization: [[paper]](https://arxiv.org/abs/1412.6980)
+*   `Keras`: Python 深度学习库：[[doc]](https://keras.io/)
+*   Adam: 一种随机优化方法：[[paper]](https://arxiv.org/abs/1412.6980)
 *   Improving neural networks by preventing co-adaptation of feature detectors: [[paper]](https://arxiv.org/abs/1207.0580)
 
-In this post, you learn how to define and evaluate accuracy of a neural network for multi-class classification using the Keras library.
-The script used to illustrate this post is provided here : [[.py](https://github.com/Cdiscount/IT-Blog/blob/master/scripts/pos_tagging_neural_nets_keras.py)|[.ipynb](https://github.com/Cdiscount/IT-Blog/blob/master/scripts/pos_tagging_neural_nets_keras.ipynb)].
+在本文中，您学习如何使用 Keras 库定义和评估用于多分类的神经网络的准确性。
+代码在这里 : [[.py](https://github.com/Cdiscount/IT-Blog/blob/master/scripts/pos_tagging_neural_nets_keras.py)|[.ipynb](https://github.com/Cdiscount/IT-Blog/blob/master/scripts/pos_tagging_neural_nets_keras.ipynb)].
```
Comment (leviding) [inline TODO1/part-of-speech-tagging-tutorial-with-the-keras-deep-learning-library.md]: 冒号换成中文的，前后不用空格，最后的句号换成中文的。

**Comment by leviding** [comment]
@luochen1992 还有些问题，再修改下吧
